task/backend/coordinator/task_coordinator.go

```diff
@@ -0,0 +1,183 @@
+package coordinator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/task/backend"
+	"github.com/influxdata/influxdb/task/backend/executor"
+	"github.com/influxdata/influxdb/task/backend/middleware"
+	"github.com/influxdata/influxdb/task/backend/scheduler"
+	"go.uber.org/zap"
+)
+
+var _ middleware.Coordinator = (*Coordinator)(nil)
+var _ Executor = (*executor.TaskExecutor)(nil)
+var _ scheduler.SchedulableService = (*SchedulableTaskService)(nil)
+
+// Executor is an abstraction of the task executor with only the functions needed by the coordinator
+type Executor interface {
+	ManualRun(ctx context.Context, id scheduler.ID, runID scheduler.ID) (executor.Promise, error)
+	Cancel(ctx context.Context, runID scheduler.ID) error
+}
+
+type TaskCoordinator struct {
+	logger *zap.Logger
+	sch    scheduler.Scheduler
+	ex     Executor
+	// UpdateTaskFunc
+
+	limit int
+}
+
+type OptionC func(*TaskCoordinator)
+
+// SchedulableTask is a wrapper around the Task struct, giving it methods to make it compatible with the Scheduler
+type SchedulableTask struct {
+	*influxdb.Task
+}
+
+// SchedulableTaskService provides an API to update the LatestScheduled time of a task
+type SchedulableTaskService struct {
+	ts influxdb.TaskService
+}
+
+// UpdateLastScheduled uses the task service to store the latest time a task was scheduled to run
+func (c *SchedulableTaskService) UpdateLastScheduled(ctx context.Context, id scheduler.ID, t time.Time) error {
+	s := t.Format(time.RFC3339Nano)
+	tid := influxdb.ID(id)
+	_, err := c.ts.UpdateTask(ctx, tid, influxdb.TaskUpdate{
+		LatestCompleted: &s,
+	})
+
+	if err != nil {
+		return fmt.Errorf("could not update last scheduled for task; Err: %v", err)
+	}
+	return nil
+}
+
+func (t SchedulableTask) ID() scheduler.ID {
+	return scheduler.ID(t.Task.ID)
+}
+
+// Schedule takes the time a Task is scheduled for and returns a Schedule object
+func (t SchedulableTask) Schedule() scheduler.Schedule {
+	// TODO (al): use new scheduler's NewSchedule method
+	return scheduler.Schedule{}
+}
+
+// Offset returns a time.Duration for the Task's offset property
+func (t SchedulableTask) Offset() (time.Duration, error) {
+	return t.OffsetDuration()
+}
+
+// LastScheduled parses the task's LatestCompleted value as a Time object
+func (t SchedulableTask) LastScheduled() (time.Time, error) {
+	return t.LatestCompletedTime()
+}
+
+func WithLimitC(i int) OptionC {
+	return func(c *TaskCoordinator) {
+		c.limit = i
+	}
+}
+
+func NewCoordinator(logger *zap.Logger, scheduler scheduler.Scheduler, executor Executor, opts ...OptionC) *TaskCoordinator {
+	c := &TaskCoordinator{
+		logger: logger,
+		sch:    scheduler,
+		ex:     executor,
+		limit:  1000,
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+// TaskCreatedC asks the Scheduler to schedule the newly created task
+func (c *TaskCoordinator) TaskCreatedC(ctx context.Context, task *influxdb.Task) error {
```

Comment:
Im not sure where `TaskCreatedC` comes from or what the last `C` means. I believe your coordinator is supposed to fulfill the interface the middleware is expecting.
https://github.com/influxdata/influxdb/blob/0cc9caa1d424ad0f8c1151feafba28f34175805a/task/backend/middleware/middleware.go#L14-L21

Reply:
The C is just added because since the two files are in the same package, it doesn't like it having the same name. I can move it to the outer directory though and that would fix it.